src/home/home.service.ts

```diff
@@ -0,0 +1,163 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { type } from 'os';
+import { Module } from 'src/entities/module.entity';
+import { Repository } from 'typeorm';
+import { User } from '../entities/user.entity'
+import { CaretakerHome, ElderlyHome, ElderlyInfo } from './interface/home.interface';
+
+@Injectable()
+export class HomeService {
+    constructor(
+        @InjectRepository(User)
+        private userRepository: Repository<User>,
+        @InjectRepository(Module)
+        private moduleRepository : Repository<Module>,
+    ) { }
+    
+    showName(user: User): string {
+        if (user.dname != null) {
+            return user.dname
+        } else {
+            return user.fname
+        }
+    }
+
+    async getElderlyHome(uid: number): Promise<ElderlyHome> {
+        const user = await this.userRepository.findOne({ uid: uid }, {
+            relations: ["modules"]
+        })
+
+        if(!user){
+            throw new HttpException("User not found", HttpStatus.NOT_FOUND)
+        }
+
+        return {
+            dname: this.showName(user),
+            imageid: user.imageid,
+            listModuleid: user.modules.map(function (module) {
+                return module.moduleid;
+            })
+        }
+    }
+
+    async getModuleList(): Promise<Module[]>{
```

Comment:
Maybe `findAllModules` or `getModules` is a better name

Reply:
I change to getAllModule